crates/sdk/macros/src/logic/method.rs

```diff
@@ -237,6 +259,20 @@ impl<'a, 'b> TryFrom<LogicMethodImplInput<'a, 'b>> for LogicMethod<'a> {
             }
         }
 
+        let name = &input.item.sig.ident;
+
+        match (is_init, &self_type, name.to_string().as_str()) {
+            (true, Some(_), _) => errors.subsume(syn::Error::new_spanned(
+                input.item,
+                errors::ParseError::NoSelfReceiverAtInit,
+            )),
+            (false, _, "init") => errors.subsume(syn::Error::new_spanned(
+                input.item,
+                errors::ParseError::InitMethodWithoutNoInitAttribute,
+            )),
+            _ => {}
```

Comment:
`Ident: PartialEq<T>` where `T: AsRef<str>`

```rust
match (is_init, &self_type) {
    (true, Some(_)) => errors.subsume(syn::Error::new_spanned(
        input.item,
        errors::ParseError::NoSelfReceiverAtInit,
    )),
    (true, None) if name != "init" => { /* forbid this */ },
    (false, _) if name == "init" => errors.subsume(syn::Error::new_spanned(
        input.item,
        errors::ParseError::InitMethodWithoutNoInitAttribute,
    )),
    _ => {}
```

Reply:
Ah, okay, we are also making sure the actual function is named init
Fixed!